runtime-modules/proposals/discussion/src/lib.rs

```diff
@@ -214,21 +240,85 @@ decl_module! {
 
             Self::ensure_thread_mode(origin, post_author_id, thread_id)?;
 
+            // Ensure account has enough funds
+            if editable {
+                ensure!(
+                    Balances::<T>::usable_balance(&account_id) >= T::PostDeposit::get(),
+                    Error::<T>::InsufficientBalanceForPost,
+                );
+            }
+
             // mutation
 
+            if editable {
+                Self::transfer_to_state_cleanup_treasury_account(
+                    T::PostDeposit::get(),
+                    thread_id,
+                    &account_id,
+                )?;
+            }
+
             let next_post_count_value = Self::post_count() + 1;
             let new_post_id = next_post_count_value;
+            let post_id = T::PostId::from(new_post_id);
 
-            let new_post = DiscussionPost {
-                author_id: post_author_id,
-            };
+            if editable {
+                let new_post = DiscussionPost {
+                    author_id: post_author_id,
+                    cleanup_pay_off: T::PostDeposit::get(),
+                    last_edited: frame_system::Module::<T>::block_number(),
+                };
+
+                <PostThreadIdByPostId<T>>::insert(thread_id, post_id, new_post);
+            }
 
-            let post_id = T::PostId::from(new_post_id);
-            <PostThreadIdByPostId<T>>::insert(thread_id, post_id, new_post);
             PostCount::put(next_post_count_value);
             Self::deposit_event(RawEvent::PostCreated(post_id, post_author_id, thread_id, text));
        }
 
+        #[weight = WeightInfoDiscussion::<T>::delete_post()]
```

Comment:
Please, change the module comment to reflect the new extrinsic (and other changes if any).